src/main/kotlin/no/nav/arbeidsplassen/importapi/adadminstatus/AdminStatusSyncWithKafka.kt

```diff
@@ -0,0 +1,69 @@
+package no.nav.arbeidsplassen.importapi.adadminstatus
+
+import io.micronaut.configuration.kafka.ConsumerAware
+import io.micronaut.configuration.kafka.annotation.*
+import io.micronaut.context.annotation.Requires
+import io.micronaut.context.annotation.Value
+import io.micronaut.core.convert.format.Format
+import no.nav.arbeidsplassen.importapi.feed.AdTransport
+import org.apache.kafka.clients.consumer.Consumer
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener
+import org.apache.kafka.common.TopicPartition
+import org.slf4j.LoggerFactory
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.util.stream.Collector
+import java.util.stream.Collectors
+
+
+@Requires(property = "adminstatussync.kafka.enabled", value="true")
+@KafkaListener(groupId="\${adminstatus.kafka.group-id:import-api-adminstatussync}", threads = 1, offsetReset = OffsetReset.EARLIEST,
+        batch = true, offsetStrategy = OffsetStrategy.SYNC)
+class AdminStatusSyncWithKafka(private val adminStatusRepository: AdminStatusRepository,
+                               @Value("\${adminstatussync.kafka.offsettimestamp}")
+                               @Format("yyyy-MM-dd'T'HH:mm:ss'Z'") private val offsetTimeStamp: LocalDateTime?): ConsumerRebalanceListener, ConsumerAware<Any,Any> {
+
+    private lateinit var consumer: Consumer<Any,Any>
+
+    companion object {
+        private val LOG = LoggerFactory.getLogger(AdminStatusSyncWithKafka::class.java)
+    }
+
+    @Topic("\${adminstatus.kafka.topic:StillingIntern}")
+    fun kakfkaAdminStatusSyncWithAd(adList: List<AdTransport>, offsets: List<Long>) {
+        LOG.info("received kafka {}", adList.size)
```

Comment:
Kan være greit å logge id'ene på annonsene her. Lettere å feilsøke hvis noe ikke dukker opp av en eller annen grunn da.

Reply:
> Kan være greit å logge id'ene på annonsene her. Lettere å feilsøke hvis noe ikke dukker opp av en eller annen grunn da.

Tror det blir for mange logg hvis batchen fylles inn med 500 på size. Det er bedre å logge når det feiler, da får du iden